route-rs-afpacket/src/input.rs

```diff
@@ -0,0 +1,70 @@
+use afpacket;
+use futures::{
+    self, ready,
+    task::{Context, Poll},
+};
+use route_rs_runtime::link;
+use std::{mem, pin::Pin};
+
+const MTU: usize = 1500;
+
+#[derive(Default)]
+pub struct AfPacketInputLink {
+    ingress: Option<afpacket::RecvHalf>,
+}
+
+impl AfPacketInputLink {
+    pub fn new() -> Self {
+        Self { ingress: None }
+    }
+
+    pub fn channel(self, ingress: afpacket::RecvHalf) -> Self {
+        Self {
+            ingress: Some(ingress),
+        }
+    }
+}
+
+impl link::LinkBuilder<(), Vec<u8>> for AfPacketInputLink {
```

Comment:
Is it possible for AfPacketInputLink to return half a packet in one recv, then the other half on the next recv? Or does it only return discreet packets?